internal/servicelog/formatter_test.go

```diff
@@ -46,6 +46,39 @@ func (s *formatterSuite) TestFormat(c *C) {
 `[1:], timeFormatRegex))
 }
 
+func (s *formatterSuite) TestTrim(c *C) {
+	raw := `
+3/4/3005 hello my name is joe
+4/5/4200 and I work in a button factory
+1/1/0033 this log entry is very old
+and dates in the middle 1/1/0033 are kept
+`[1:]
+
+	chunkSizes := []int{1, 2, 3, 4, 5, 6, 7, 11, 13, 27, 100}
+	for _, size := range chunkSizes {
+		b := &bytes.Buffer{}
+		w, err := servicelog.NewTrimWriter(b, "[0-9]{1,2}/[0-9]{1,2}/[0-9]{4} +")
+		if err != nil {
+			c.Fatal(err)
+		}
+		c.Logf("Chunk size %v\n", size)
+		for pos := 0; pos < len(raw); pos += size {
+			end := pos + size
+			if end > len(raw) {
+				end = len(raw)
+			}
+			fmt.Fprint(w, raw[pos:end])
+		}
+
+		c.Assert(b.String(), Equals, fmt.Sprintf(`
+hello my name is joe
+and I work in a button factory
+this log entry is very old
+and dates in the middle 1/1/0033 are kept
+`[1:]))
+	}
+}
```

Comment:
Are we getting full coverage with this one test? Even if so, seems like we should probably add a few more edge case tests: one with no writes, one with blank lines, one where the last line doesn't have a `\n` in it.

Reply:
Should be pretty decent coverage, but you're right - I'll work on some more tests.

Reply:
I rounded out these tests a bit and checked coverage - we're basically hitting everything now.